src/dmclock/src/dmclock_server.h

```diff
@@ -296,9 +296,9 @@ namespace crimson {
 	// an idle client becoming unidle
 	double                prop_delta = 0.0;
 
-	c::IndIntruHeapData   reserv_heap_data;
-	c::IndIntruHeapData   lim_heap_data;
-	c::IndIntruHeapData   ready_heap_data;
+	c::IndIntruHeapData   reserv_heap_data{};
```

Comment:
Huh? If they can be default-initialized, why is this change needed?

Reply:
strictly speaking, they are not "default-initialized". what is used here is "[zero-initialization](http://en.cppreference.com/w/cpp/language/zero_initialization)". we could use it without doing so explicitly, but it depends on how `ClientRec` is initialized. if we initialize it using "zero initialization", like

```c++
auto c = ClientRec{};
```
using the default ctor, then `{reserv,lim,ready}_heap_data`  are zero-initialized. see http://en.cppreference.com/w/cpp/language/value_initialization. but `ClientRec` has a user-defined ctor. so one way to initialize them is to use the "zero initialization" explicitly.

Reply:
@amitkumar50 please read https://github.com/ceph/ceph/blob/master/src/dmclock/README.before-modifying-files-here-or-below and resubmit this change.

Reply:
Didn't realize it was just a typedef for ```size_t``` -- this is an excellent example of why this style of initialization is confusing.

Reply:
i think the nice thing is that it's "confusing". because `IndIntruHeapData` is a typedef, i guess @ivancich wanted to hide the actual type of `IndIntruHeapData` from its users, so we can change it to other types (e.g, compound types) when necessary. in that case, the notation of `{}` will still work if they offer a default ctor.

Reply:
@tchaikov Thank you for that input. In my commit to the dmclock library to address this very same issue initialized with "= 0". But your point is well-taken and I'll look into changing it to "{}".